wcfsetup/install/files/acp/update_com.woltlab.wcf_5.4_migrate_rank_images.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use wcf\data\user\rank\UserRank;
+use wcf\data\user\rank\UserRankEditor;
+use wcf\data\user\rank\UserRankList;
+use wcf\system\background\BackgroundQueueHandler;
+use wcf\system\background\job\DownloadRankImageJob;
+use wcf\util\Url;
+
+/**
+ * Downloads the rank image files to the new internal upload system.
+ *
+ * @author	Joshua Ruesweg
+ * @copyright	2001-2021 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core
+ */
+$rankList = new UserRankList();
+$rankList->readObjects();
+
+foreach ($rankList as $rank) {
+	if (!empty($rank->rankImage)) {
+		if (Url::is($rank->rankImage)) {
+			BackgroundQueueHandler::getInstance()->enqueueIn(new DownloadRankImageJob($rank), 60);
+			continue;
+		}
+		
+		$rankEditor = new UserRankEditor($rank);
+		
+		if (file_exists(WCF_DIR . $rank->rankImage)) {
+			$newImageName = WCF_DIR . UserRank::RANK_IMAGE_DIR . $rank->rankID .'-'. basename($rank->rankImage);
+			copy(WCF_DIR . $rank->rankImage, $newImageName);
```

Comment:
This should probably handle failures somewhat gracefully to prevent the upgrade from making progress.